nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/manager/ControllerServiceEntityMerger.java

```diff
@@ -36,6 +36,17 @@
 
 public class ControllerServiceEntityMerger implements ComponentEntityMerger<ControllerServiceEntity> {
 
+    @Override
+    public void merge(ControllerServiceEntity clientEntity, Map<NodeIdentifier, ControllerServiceEntity> entityMap) {
+        ComponentEntityMerger.super.merge(clientEntity, entityMap);
+        for (Map.Entry<NodeIdentifier, ControllerServiceEntity> entry : entityMap.entrySet()) {
+            final ControllerServiceEntity entityStatus = entry.getValue();
+            if (clientEntity != entityStatus) {
```

Comment:
This looks like it should use `clientEntity.equals()` instead of the instance comparison.

Reply:
In this case, it may be find to use .equals()..  it would mean we will not bother doing a StatusMerge if the statuses are the same, which seems logical.

Reply:
As noted on the `ReportingTaskEntity`, `ControllerServiceEntity` does not override `equals()`, so keeping the check as it stands sounds good.